src/main/java/com/google/cloud/tools/managedcloudsdk/ChildProgressListener.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.managedcloudsdk;
+
+/** Default implementation of a child listener. Should satisfy the simple use case. */
+public class ChildProgressListener implements ProgressListener {
+
+  private final ProgressListener parent;
+
+  private long totalWork;
+  private long totalWorkDone = 0;
+
+  private final long totalAllocatedWork;
+  private long totalAllocatedWorkDone = 0;
+
+  /**
+   * Create a child progress listener that performs a subtask for a parent.
+   *
+   * @param parent the parent ProgressListener that created this child
+   * @param totalAllocatedWork the amount of work from the parent this child is expected to report.
+   */
+  public ChildProgressListener(ProgressListener parent, long totalAllocatedWork) {
+    this.parent = parent;
+    this.totalAllocatedWork = totalAllocatedWork;
+  }
+
+  @Override
+  public void start(String message, long totalWork) {
+    parent.update(message);
+    this.totalWork = totalWork;
+  }
+
+  @Override
+  public void update(long workDone) {
+    totalWorkDone += workDone;
+
+    if (totalWork == UNKNOWN) {
+      // there is room here for an implementation of an unknown amount of work
+      // algorithm, but for now we just ignore any update.
+      return;
+    }
+
+    // normalized work done since last update
+    long normalizedWorkDone =
```

Comment:
nit: might be clearer to put ` - totalAllocatedWorkDone` on in the update on L62, so that the logic is

1) new total allocated work done = work done normalized to total allocated work
2) update parent with difference between new and old
3) update old to new

Reply:
Or, perhaps the allocation shouldn't be visible to the child, but rather the parent manages that, since the parent is the one that creates the child via `newChild`. But, that would require changing the `ProgressListener` API.

Reply:
The logic as it is right now accounts for the differences in increments between `allocation` and child `total work`. So if `workDone < single unit of allocation` then we have to make sure we carry that over. But I'm not actually sure if that's what you're talking about.

I think the parent shouldn't care what the child is doing with its allocation. As long as it does `allocation` amount of work. I think if we move that translation up to the parent, it would have to keep state.